src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -62,21 +62,21 @@ private void AddSpecificNodesSuppressOperations(List<SuppressOperation> list, Sy
 
             if (node is MemberDeclarationSyntax memberDeclNode)
             {
-                // Attempt to keep the part of a member that follows hte attributes on a single
+                // Attempt to keep the part of a member that follows the attributes on a single
                 // line if that's how it's currently written.
                 var tokens = memberDeclNode.GetFirstAndLastMemberDeclarationTokensAfterAttributes();
                 AddSuppressWrappingIfOnSingleLineOperation(list, tokens.Item1, tokens.Item2);
 
+                // Also, If the member is on single line with its attributes on it, then keep 
+                // it on a single line.  This is for code like the following:
+                //
+                //      [Import] public int Field1;
+                //      [Import] public int Field2;
                 var attributes = memberDeclNode.GetAttributes();
-                if (attributes.Count > 0)
+                for (int i = 0; i < attributes.Count; ++i)
                 {
-                    // Also, If the member is on single line with its attributes on it, then keep 
-                    // it on a single line.  This is for code like the following:
-                    //
-                    //      [Import] public int Field1;
-                    //      [Import] public int Field2;
                     AddSuppressWrappingIfOnSingleLineOperation(list,
```

Comment:
❗️ We should really have a `break;` statement for after the first case where a suppression is actually added to `list` by this call. Otherwise we add a sequence of overlapping suppressions, which doesn't seem to be causing a problem now but I also don't know that it was intended to handle that case.

Reply:
This would require to detect whether two tokens are on the same line. The formatter itself does this via [`_tokenStream.TwoTokensOriginallyOnSameLine`](https://github.com/MaStr11/roslyn/blob/PreserveAttributesOnSingleLine/src/Workspaces/Core/Portable/Formatting/Context/FormattingContext.cs#L261). I have no idea how to get the TokenStream from a SyntaxToken.

Reply:
I think [`AddInitializerSuppressOperations`](https://github.com/MaStr11/roslyn/blob/master/src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs#L212) adds overlapping suppressions too.

Reply:
❓ Can you not just check to see if anything was added?

```csharp
var originalListCount = list.Count;
AddSuppressWrappingIfOnSingleLine(...);
if (list.Count > originalListCount)
{
  // Any additional spans added by this loop would just be sub-spans of the one just added
  break;
}
```

Reply:
The suppression is  added (more or less) unconditionally. Checking the list count does not help as count changes no matter if the tokens are on the same line or not. (Adding is only skipped if `startToken.Kind() == SyntaxKind.None || endToken.Kind() == SyntaxKind.None`, which is never the case in the loop).

Reply:
Thanks for the update. It sounds from your earlier investigation that the duplicates won't hurt. We'll go with that approach.